mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2320,12 +2320,25 @@ impl UtxoRpcClientOps for ElectrumClient {
     }
 
     fn display_balance(&self, address: Address, decimals: u8) -> RpcRes<BigDecimal> {
-        let hash = electrum_script_hash(&output_script(&address, ScriptType::P2PKH));
-        let hash_str = hex::encode(hash);
-        Box::new(
-            self.scripthash_get_balance(&hash_str)
-                .map(move |electrum_balance| electrum_balance.to_big_decimal(decimals)),
-        )
+        let this = self.clone();
+        let mut hashes = Vec::new();
+        // If the plain pubkey is available, fetch the balance found in P2PK as well (if any).
+        if address.hash.is_pubkey() {
+            hashes.push(hex::encode(electrum_script_hash(&output_script(&address, ScriptType::P2PK))));
+        }
+        hashes.push(hex::encode(electrum_script_hash(&output_script(&address, ScriptType::P2PKH))));
+
+        let fut = async move {
+            Ok(this
+                .scripthash_get_balances(hashes)
+                .compat()
+                .await?
+                .into_iter()
+                .fold(BigDecimal::from(0), |sum, electrum_balance| {
+                    sum + electrum_balance.to_big_decimal(decimals)
+                }))
```

Comment:
It's a bit of an overhead to send 2 requests to electrums to get both `P2PK` and `P2PKH` balances when most addresses will not have `P2PK` outputs but I don't see a way around it. Just opening this for discussions in case you or anybody else has a better idea.
P.S. we opted for this solution as any other solution we could find would lead to a new coin for `P2PK` addresses, some explorers do the same as it was pointed out in a note in the code changes but others don't. A complete solution would be to have only segwit addresses for any coin that supports it and include all kind of outputs in the balance, but this is outside the scope of this issue/PR and more related to this https://github.com/KomodoPlatform/komodo-defi-framework/issues/2050. This solution would lead to having balance on addresses completely different from explorers though.